### ISSUE TITLE
Constraint with FixedPolynomialBasis?

### ISSUE THREAD

**Issue by AaronMoat** [JuliaOpt/SumOfSquares.jl#159]
Hi there,

I was wondering if there's a way to omit certain monomials from the default constraint basis or specify a custom basis? For example:

```julia
using DynamicPolynomials, SumOfSquares, MosekTools

model = SOSModel(Mosek.Optimizer)

@polyvar x
p = x^4 + 17

@variable(model, gamma)
@objective(model, Max, gamma)

constr = @constraint(model, p - gamma >= 0, basis = FixedPolynomialBasis([x^2]))

optimize!(model)

@show objective_value(model)
@show gram_matrix(constr).basis
```

Omitting the basis uses the default of `[1, x, x^2]` which is unnecessary here. The above gives a type error:

```
ERROR: LoadError: MethodError: no method matching SumOfSquares.Certificate.Newton(::NonnegPolyInnerCone{MathOptInterface.PositiveSemidefiniteConeTriangle}, ::FixedPolynomialBasis{Monomial{true},Array{Monomial{true},1}}, ::Tuple{})
Closest candidates are:
  SumOfSquares.Certificate.Newton(::CT, !Matched::Type{BT}, ::NPT) where {CT<:SumOfSquares.SOSLikeCone, BT<:AbstractPolynomialBasis, NPT<:Tuple} at /Users/aaronmoat/.julia/packages/SumOfSquares/a0iRj/src/Certificate/Certificate.jl:131
```

If there's no way, are there any workarounds?

**Comment by blegat** [comment]
Thanks for raising this issue, it should be fixed by https://github.com/JuliaOpt/SumOfSquares.jl/pull/160

**Comment by blegat** [comment]
It should be fixed in v0.4.1 just released. Please reopen if that's not the case.
